lib/ocpp/common/websocket/websocket_tls.cpp

```diff
@@ -171,33 +171,36 @@ tls_context WebsocketTLS::on_tls_init(std::string hostname, websocketpp::connect
         }
 
         if (security_profile == 3) {
-            const auto certificate_key_pair =
-                this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
+            const auto certificate_info =
+                this->evse_security->get_leaf_certificate_info(CertificateSigningUseEnum::ChargingStationCertificate);
 
-            if (certificate_key_pair.has_value() && certificate_key_pair.value().password.has_value()) {
-                std::string passwd = certificate_key_pair.value().password.value();
+            if (certificate_info.has_value() && certificate_info.value().password.has_value()) {
+                std::string passwd = certificate_info.value().password.value();
                 context->set_password_callback(
                     [passwd](auto max_len, auto purpose) { return passwd.substr(0, max_len); });
             }
 
-            if (!certificate_key_pair.has_value()) {
+            if (!certificate_info.has_value()) {
                 EVLOG_AND_THROW(std::runtime_error(
                     "Connecting with security profile 3 but no client side certificate is present or valid"));
             }
```

Comment:
This could be moved up (L176) so that you dont need to check for `!certificate_info.has_value()` multiple times

Reply:
The update has not been done for the deprecated websocket, will be made for backwards-compatibility.